internal/providers/nscloud/api/rpc.go

```diff
@@ -276,8 +265,14 @@ func ListClusters(ctx context.Context, api API) (*KubernetesClusterList, error)
 }
 
 func ExchangeToken(ctx context.Context, scopes ...string) (string, error) {
+	// Check if there is already tenant token stored.
 	tenantToken, err := auth.LoadTenantToken()
 	if err == nil {
+		// If no scopes provided we can immediately return the token.
+		if len(scopes) == 0 {
+			return tenantToken.TenantToken, nil
+		}
+
 		resp, err := fnapi.ExchangeTenantToken(ctx, tenantToken.TenantToken, scopes)
```

Comment:
Ideally calls that have non-trivial duration should be wrapped in tasks, so users can see where time is being spent.

Reply:
done. I hope simply wrapping the whole func body is OK.